### PR TITLE
Fix xcmp message processing condition.

### PR DESCRIPTION
**Pull request by shaunxw**
The messages processing loop shouldn't be entered again if there is no message left. Before this fix, calling `handle_xcmp_messages` with one message would panic. https://github.com/paritytech/cumulus/blob/1bb5e2ae4ba91647e2c26b5a820211a7274ed53a/pallets/xcmp-queue/src/lib.rs#L609 https://github.com/paritytech/cumulus/blob/1bb5e2ae4ba91647e2c26b5a820211a7274ed53a/pallets/xcmp-queue/src/lib.rs#L522-L528

Fixes: https://github.com/paritytech/cumulus/issues/473

### CODE REVIEW

**Comment by apopiak** [comment]
add a smoke test?

**Comment by bkchr** [comment]
> add a smoke test?

https://github.com/paritytech/cumulus/issues/400

@apopiak can you review this?

**Review by gavofyork — APPROVED** [review]

**Comment by gavofyork** [comment]
A test would be nice.

**Comment by apopiak** [comment]
> > add a smoke test?
> 
> #400
> 
> @apopiak can you review this?

@bkchr Do you mean that it's fine to merge as-is without a test because we have a tracking issue?

**Comment by shawntabrizi** [comment]
@apopiak no we should add a test here. If you can, add one, if not, I will do it later today

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
@apopiak please take a look and merge if you think it is okay

File: pallets/xcmp-queue/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+pub struct XcmConfig;
+impl xcm_executor::Config for XcmConfig {
+	type Call = Call;
+	type XcmSender = XcmRouter;
+	// How to withdraw and deposit an asset.
+	type AssetTransactor = LocalAssetTransactor;
+	type OriginConverter = ();
+	type IsReserve = NativeAsset;
+	type IsTeleporter = NativeAsset; // <- should be enough to allow teleportation of ROC
```
Comment (apopiak) [inline pallets/xcmp-queue/src/mock.rs:141]:
```suggestion
	type IsTeleporter = NativeAsset;
```

File: pallets/xcmp-queue/src/mock.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	type LocationInverter = LocationInverter<Ancestry>;
+	type Barrier = ();
+	type Weigher = FixedWeightBounds<UnitWeightCost, Call>;
+	type Trader = ();
+	type ResponseHandler = (); // Don't handle responses for now.
```
Comment (apopiak) [inline pallets/xcmp-queue/src/mock.rs:146]:
```suggestion
	type ResponseHandler = ();
```

**Review by apopiak — APPROVED** [review]

**Pull request merged**